tests/extensions/test_extension_python.py

```diff
@@ -31,3 +34,57 @@ def test_grouping_and_choice_failures(test_file, expected_exit_code, expected_me
 
     if expected_exit_code == EXIT_CODE_FAILURE:
         assert "Search failed due to linting errors!" in result.output
+
+
+@pytest.mark.parametrize('test_file,annotations', [
```

Comment:
Nice.  We always seem to use `ddt` for this.  I wonder if that is just historical from using nose?

Reply:
Probably. I just replicated the pattern I saw elsewhere in the other unit tests.
As a side note, I must say I'm not a big fan of pytest. The output seems really verbose to me, and weirdly formatted. But maybe that's just me and I need to get more used to it.